public/static/app/perfdiscovery/PerfDiscoveryService.js

```diff
@@ -161,18 +161,23 @@ mciModule.factory('PerfDiscoveryService', function($q, ApiV1, ApiTaskdata) {
         var appendix = {
           ratio: ratio(item.speed, baseline.speed),
           baseSpeed: baseline.speed,
-          trendData: _.chain(results.history)
-            .take(100)
-            .map(function(run) {
-              var base = run[id]
-              if (base == undefined || !base.speed) {
-                return 1
-              }
-              return ratio(item.speed, base.speed)
-            })
-            .value(),
         }
-        var avgRatio = d3.mean(appendix.trendData)
+
+        var trendData = _.chain(results.history)
+          .take(100)
+          .map(function(run) {
+            var base = run[id]
+            if (base == undefined || !base.speed) {
```

Comment:
I think this is wrong.

An undefined base or base.speed should result in an undefined entry in the list.
A 0 base.speed should result in a Nan.